weave_query/tests/test_wb.py

```diff
@@ -1776,16 +1776,66 @@ def test_is_valid_version_string():
-    name = 12347187287418787843872388177814
-    path = "table #3.table.json"
-    result = wb_util.escape_artifact_path(
-        f"wandb-client-artifact://{name}:latest/{path}"
-    )
-    uri = artifact_wandb.WeaveWBLoggedArtifactURI.parse(result)
-
-    assert uri.path == path
+class TestEscapeArtifactPath:
+    wandb_prefix = "wandb-client-artifact://"
+    artifact_name = "12347187287418787843872388177814"
+    file_path = "table #3.table.json"
+    file_path_with_colon = "(batch: 1) table #3.table.json"
+    version = "latest"
+    # version_with_colon = "latest:v0"
+
+    def test_standard_artifact_path(self):
+        """
+        Artifact path of the form: "<name>:<version>/<path>" should be escaped properly.
+        """
+        result = wb_util.escape_artifact_path(
+            f"{self.wandb_prefix}{self.artifact_name}:{self.version}/{self.file_path}"
+        )
+        uri = artifact_wandb.WeaveWBLoggedArtifactURI.parse(result)
+        assert uri.path == self.file_path
+
+    def test_artifact_path_with_no_version(self):
+        """
+        Artifact path of the form: "<name>/<path>" should be escaped properly.
+        """
+        result = wb_util.escape_artifact_path(
+            f"{self.wandb_prefix}{self.artifact_name}/{self.file_path}"
+        )
+        uri = artifact_wandb.WeaveWBLoggedArtifactURI.parse(result)
+        assert uri.path == self.file_path
+
+    def test_artifact_path_with_version_and_file_with_colon(self):
+        """
+        Artifact path of the form: "<name>:<version>/<path>" should be escaped properly,
+        where <path> contains a colon.
+        """
+        result = wb_util.escape_artifact_path(
+            f"{self.wandb_prefix}{self.artifact_name}:{self.version}/{self.file_path_with_colon}"
+        )
+        uri = artifact_wandb.WeaveWBLoggedArtifactURI.parse(result)
+        assert uri.path == self.file_path_with_colon
+
+    def test_artifact_path_with_no_version_and_file_with_colon(self):
+        """
+        Artifact path of the form: "<name>/<path>" should be escaped properly,
+        where <path> contains a colon.
+        """
+        result = wb_util.escape_artifact_path(
+            f"{self.wandb_prefix}{self.artifact_name}/{self.file_path_with_colon}"
+        )
+        uri = artifact_wandb.WeaveWBLoggedArtifactURI.parse(result)
+        assert uri.path == self.file_path_with_colon
+
+    # def test_artifact_path_with_version_with_colon_and_file_with_colon(self):
+    #     """
+    #     Artifact path of the form: "<name>:<version>/<path>" should be escaped properly,
+    #     where <version> contains a colon and <path> contains a colon.
+    #     """
+    #     result = wb_util.escape_artifact_path(
+    #         f"{self.wandb_prefix}{self.artifact_name}:{self.version_with_colon}/{self.file_path_with_colon}"
+    #     )
+    #     uri = artifact_wandb.WeaveWBLoggedArtifactURI.parse(result)
+    #     assert uri.path == self.file_path_with_colon
```

Comment:
The resulting URI `wandb-client-artifact://12347187287418787843872388177814:latest:v0/%28batch%3A%201%29%20table%20%233.table.json` is not actually valid (We run into a `errors.WeaveInvalidURIError`)

Reply:
what is the error this raises? can you share it here?

Reply:
Yup, here it is:
```
weave_query/artifact_wandb.py:1087: in parse
    return cls.from_parsed_uri(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'weave_query.artifact_wandb.WeaveWBLoggedArtifactURI'>
uri = 'wandb-client-artifact://12347187287418787843872388177814:latest:v0/%28batch%3A%201%29%20table%20%233.table.json', schema = 'wandb-client-artifact'
netloc = '12347187287418787843872388177814:latest:v0', path = '(batch: 1) table #3.table.json', params = '', query = {}, fragment = ''

    @classmethod
    def from_parsed_uri(
        cls,
        uri: str,
        schema: str,
        netloc: str,
        path: str,
        params: str,
        query: dict[str, list[str]],
        fragment: str,
    ):
        path = parse.unquote(path.strip("/"))
        spl_netloc = netloc.split(":")
        if len(spl_netloc) == 1:
            name = spl_netloc[0]
            version = None
        elif len(spl_netloc) == 2:
            name, version = spl_netloc
        else:
>           raise errors.WeaveInvalidURIError(f"Invalid WB Client Artifact URI: {uri}")
E           weave_query.errors.WeaveInvalidURIError: Invalid WB Client Artifact URI: wandb-client-artifact://12347187287418787843872388177814:latest:v0/%28batch%3A%201%29%20table%20%233.table.json

weave_query/artifact_wandb.py:1041: WeaveInvalidURIError
```

Reply:
I think this should work if we make `spl_netloc = netloc.split(':', 1)`